opentelemetry-sdk/src/opentelemetry/sdk/logs/severity.py

```diff
@@ -0,0 +1,100 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import enum
+
+
+class SeverityNumber(enum.Enum):
+    UNSPECIFIED = 0
+    TRACE = 1
+    TRACE2 = 2
+    TRACE3 = 3
+    TRACE4 = 4
+    DEBUG = 5
+    DEBUG2 = 6
+    DEBUG3 = 7
+    DEBUG4 = 8
+    INFO = 9
+    INFO2 = 10
+    INFO3 = 11
+    INFO4 = 12
+    WARN = 13
+    WARN2 = 14
+    WARN3 = 15
+    WARN4 = 16
+    ERROR = 17
+    ERROR2 = 18
+    ERROR3 = 19
+    ERROR4 = 20
+    FATAL = 21
+    FATAL2 = 22
+    FATAL3 = 23
+    FATAL4 = 24
+
+
+_STD_TO_OTEL_MAP = {
+    10: SeverityNumber.DEBUG,
+    11: SeverityNumber.DEBUG2,
+    12: SeverityNumber.DEBUG3,
+    13: SeverityNumber.DEBUG4,
+    14: SeverityNumber.DEBUG4,
+    15: SeverityNumber.DEBUG4,
+    16: SeverityNumber.DEBUG4,
+    17: SeverityNumber.DEBUG4,
+    18: SeverityNumber.DEBUG4,
+    19: SeverityNumber.DEBUG4,
```

Comment:
> This piece of code is not related to reverse mapping, I didn't came across the need to use reverse mapping yet. This is used for mapping from python to OTLP format https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#mapping-of-severitynumber.

Ok, I see. What I feel arbitrary here is the multiple mapping of some integers to the same severity number (like 13, 14, 15... which are all mapped to `DEBUG4`). For this particular mapping between standard Python log levels to severity numbers, I think it makes more sense for the mapping function to accept only the [numeric values](https://docs.python.org/3/library/logging.html#logging-levels) of the standard logging levels instead.

BTW, it seems like this mapping function will be necessary for other third-party logging libraries as well? If that is the case, I think we should entry point this function too.

Reply:
It may look arbitrary but it is not. Multiple severity numbers are mapped to same severity numbers because the gap between levels is 10 in python standard but whereas it 5 in OTLP. The higher number the more severity; levelno 13-19 may mean different severity in the context of python but when they should be mapped to OTLP we assign the max possible which is `DEBUG4`.